lib/_stream_writable.js

```diff
@@ -484,83 +468,66 @@ function afterWrite(stream, state, count, cb) {
 
 // If there's something in the buffer waiting, then invoke callbacks.
 function errorBuffer(state, err) {
-  if (state.writing || !state.bufferedRequest) {
+  if (state.writing) {
     return;
   }
 
-  for (let entry = state.bufferedRequest; entry; entry = entry.next) {
-    const len = state.objectMode ? 1 : entry.chunk.length;
-    state.length -= len;
-    entry.callback(err);
+  for (const { callback } of state.buffered.slice(state.bufferedIndex)) {
```

Comment:
I would prefer this to be a for(;;) loop and avoid the generator.

Reply:
Fixed